src/profile-logic/marker-data.js

```diff
@@ -388,8 +388,12 @@ export function mergeStartAndEndNetworkMarker(
           };
           filteredMarkers.push(mergedMarker);
           i++;
+          continue;
+        }
+        // Skip duplicated markers
+        if (Object.is(marker, markerNext)) {
```

Comment:
`Object.is` is a bit more accurate.
![zetnr](https://user-images.githubusercontent.com/562354/47431132-7c4b1c00-d79b-11e8-8b28-edf11d640486.png)

Especially as we are testing the duplicated markers here and exclude those that are the same, I'd like to be as accurate as possible. I know, none of the expected values should contain a `NaN` or `+0 vs -0`, but this is not the scope of this function.

But maybe I'm also too careful here.

Reply:
Well `marker` and `markerNext` are either objects or `undefined`. And actually `markerNext` can't be undefined because of the `if` condition above, and `marker` can't be undefined because it would break in line 353 (but it can't happen anyway). So we know they're both objects, coming from the same array, so it should be strictly equivalent to `===` or even `==` (if we didn't ban that one :-) ).

Moreover I'm a bit concerned about the performance of `Object.is` compared to `===`, because we're in a hot loop here. But maybe it's performant, I just don't know.

About the algorithm itself, I don't understand how we can get the exact same objects in 2 different locations of the array (this is what this condition tests). I didn't read the rest of the code so it's possible this is just me not understanding the big picture, but a better comment could help understand this better.

Reply:
If we're adding this code solely because of tests, maybe the tests should be fixed instead.